lib/web_ui/lib/src/engine/canvaskit/noto_font.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+class NotoFont {
+  const NotoFont(this.name, this.url, this._rangeStarts, this._rangeEnds);
+
+  factory NotoFont.fromFlatRanges(String name, String url, List<int> flatRanges) {
+    final List<int> starts = <int>[];
+    final List<int> ends = <int>[];
+    for (int i = 0; i < flatRanges.length; i += 2) {
+      starts.add(flatRanges[i]);
+      ends.add(flatRanges[i+1]);
+    }
+    return NotoFont(name, url, starts, ends);
+  }
+
+  final String name;
+  final String url;
+  // A sorted list of Unicode range start points.
+  final List<int> _rangeStarts;
+
+  // A sorted list of Unicode range end points.
+  final List<int> _rangeEnds;
+
+  List<CodeunitRange> get unicodeRanges {
```

Comment:
nit: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#getters-feel-faster-than-methods

Maybe make it a `computeUnicodeRanges()` method?